### PR TITLE
V2 dependency updates

### PR DESCRIPTION
**Pull request by MynockSpit**
The only backwards-incompatibilities I found were some basic changes relating to Semantic UI. That said, I also ...

- upgraded Swagger UI, which might be a controversial change. We should take a look and see what we think.
- wrote a silly build script that lets you test the ***site code*** of the dev portal locally. You still need to deploy to test the lambda functions (though SAM might cover that).
- updated lambda function to the node 8.10 runtime. The code seems fine, and light testing surfaced no issues.
- removed the dev-portal node_modules directory from the committed assets (we already commit the build assets as part of the lambda deployer, so no need for the source).

### CODE REVIEW

File: dev-portal/package.json [hunk 1]
```diff
@@ -1,25 +1,33 @@
 {
-  "name": "dev-portal",
```
Comment (Trial-In-Error) [inline dev-portal/package.json]: I feel like we should keep it named dev-portal and bump the version (next line)?

File: dev-portal/package.json [hunk 2]
```diff
@@ -1,25 +1,33 @@
 {
```
Comment (Trial-In-Error) [inline dev-portal/package.json]: How many deprecation / npm security warns / etc. are left on `npm install`?

File: dev-portal/public/config.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+window.config = {
```
Comment (Trial-In-Error) [inline dev-portal/public/config.js]: Don't commit this file.

File: dev-portal/package.json (hunk 1 above)
Comment (MynockSpit) [reply dev-portal/package.json]: Oh, whoops. Remnant of upgrading. Will do.

File: dev-portal/public/index.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
     <!-- <meta name="viewport" content="width=device-width, initial-scale=1"> -->
     <link rel="shortcut icon" href="%PUBLIC_URL%/favicon.ico">
 
+    <script type="text/javascript" src="%PUBLIC_URL%/config.js"></script>
```
Comment (Trial-In-Error) [inline dev-portal/public/index.html]: Yep.

File: dev-portal/src/components/App/index.js [hunk 5]
```diff
@@ -53,8 +53,8 @@ export default class App extends PureComponent {
     // We are using an S3 redirect rule to prefix the url path with #!
     // This then converts it back to a URL path for React routing
     // NOTE: For local development, you will get a Page Not Found when refreshing the Swagger UI page when it has a #!
-    const hashRoute = location.hash.substring(2)
-    history.pushState({}, 'home page', hashRoute)
+    // const hashRoute = window.location.hash.substring(2)
```
Comment (Trial-In-Error) [inline dev-portal/src/components/App/index.js]: What's this do?

File: package.json [hunk 6]
```diff
@@ -6,11 +6,8 @@
   "config": {
     "marketplaceSubscriptionTopic": "arn:aws:sns:us-east-1:287250355862:aws-mp-subscription-notification-PRODUCT_CODE"
   },
-  "scripts": {
-  },
+  "scripts": {},
```
Comment (Trial-In-Error) [inline package.json]: +1

File: dev-portal/package.json (hunk 2 above)
Comment (MynockSpit) [reply dev-portal/package.json]: Zero. We've got one peer dependency warning, but it doesn't seem to be affecting anything.

File: dev-portal/src/pages/ApiDetails/index.js [hunk 7]
```diff
@@ -1,28 +1,47 @@
...
-    super(props)
-    this.state = {}
-
-    getApi(props.match.params.apiId)
+  componentDidMount = () => {
+    getApi(this.props.match.params.apiId)
     .then(api => {
-      this.setState({
-        api
+      console.log(api)
```
Comment (Trial-In-Error) [inline dev-portal/src/pages/ApiDetails/index.js]: Do we still need this?

File: dev-portal/public/swagger-ui/dist/css/print.css [hunk 8]
```diff
@@ -1,1367 +0,0 @@
-/* Original style from softwaremaniacs.org (c) Ivan Sagalaev <Maniac@SoftwareManiacs.Org> */
```
Comment (Trial-In-Error) [inline dev-portal/public/swagger-ui/dist/css/print.css]: We should try to split commits that remove assets/dependencies/libraries and that make changes (this commit contains the changes to move to swagger ui 3 as well as removing a bunch of swagger ui 2 files).

File: dev-portal/build.config.js [hunk 9]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  stackName: `nathaniels-dev-portal`,
```
Comment (Trial-In-Error) [inline dev-portal/build.config.js]: Don't commit this file.

File: dev-portal/src/components/App/index.js (hunk 5 above)
Comment (MynockSpit) [reply dev-portal/src/components/App/index.js]: No idea. It seems to work without it and break with it. My plan was to remove it and if we saw anything weird when developing further, investigate.

File: dev-portal/src/pages/ApiDetails/index.js (hunk 7 above)
Comment (MynockSpit) [reply dev-portal/src/pages/ApiDetails/index.js]: Nope. Will remove.

File: dev-portal/public/swagger-ui/dist/css/print.css (hunk 8 above)
Comment (MynockSpit) [reply dev-portal/public/swagger-ui/dist/css/print.css]: Don't agree. That would mean the step between two single commits is totally broken.

File: dev-portal/package.json (hunk 2 above)
Comment (Trial-In-Error) [reply dev-portal/package.json]: Sweet!

File: dev-portal/src/components/App/index.js (hunk 5 above)
Comment (Trial-In-Error) [reply dev-portal/src/components/App/index.js]: Yikes.

File: dev-portal/public/config.js (hunk 3 above)
Comment (MynockSpit) [reply dev-portal/public/config.js]: Yup. Removing.

File: dev-portal/public/swagger-ui/dist/css/print.css (hunk 8 above)
Comment (Trial-In-Error) [reply dev-portal/public/swagger-ui/dist/css/print.css]: That's fine. I'd prefer reviewer ease over 'every commit is releasable'. Releasable commits should be tagged, other commits should be assumed to not be guaranteed to work.

Comment (MynockSpit) [reply dev-portal/public/swagger-ui/dist/css/print.css]: :+1:

**Pull request merged**